straxen/plugins/online_monitor.py

```diff
@@ -254,24 +254,130 @@ def area_width_hist(self, data):
+        help='Number of bins of histogram of online monitor. Will be used '
+             'for: '
+             'drift_time_vs_R2-histogram, '),
+)
+class OnlineEventMonitor(strax.Plugin):
+    """
+    Plugin to write data to the online-monitor. Data that is written by
+    this plugin should be small such as to not overload the runs-
+    database.
+
+    This plugin takes 'event_basics'.
+
+    Produces 'online_event_monitor' with info on the events.
+    """
+    depends_on = ('event_basics',)
+    provides = 'online_event_monitor'
+    __version__ = '0.0.1'
+    # TODO make new datakind:
+    # data_kind = 'online_monitor'
+    rechunk_on_save = False
+    
+    def infer_dtype(self):
+        n_bins = self.config['online_monitor_nbins']
+        bounds_drift_time_R2 = self.config['drift_time_vs_R2_bounds']
+
+        dtype = [
+            (('Start time of the chunk', 'time'),
+             np.int64),
+            (('End time of the chunk', 'endtime'),
+             np.int64),
+            (('Drift time vs R^2 histogram', 'drift_time_vs_R2_hist'),
+             (np.int64, (n_bins, n_bins))),
+            (('Drift time vs R^2 edges (linear space)', 'drift_time_vs_R2_bounds'),
+             (np.float64, np.shape(bounds_drift_time_R2))),
+        ]
+        return dtype
+    
+    def compute(self, events, start, end):
+        # General setup
+        res = np.zeros(1, dtype=self.dtype)
+        res['time'] = start
+        res['endtime'] = end
+        
+        
+        # --- Drift time vs R^2 histogram ---
+        res['drift_time_vs_R2_bounds'] = self.config['drift_time_vs_R2_bounds']
+        
+        mask = self._config_as_selection_str(
+            self.config['drift_time_vs_R2_cut_string'], events)
+        res['drift_time_vs_R2_hist'] = self.drift_time_R2_hist(events[mask])
+        # Make a new mask, don't re-use
+        del mask
+        
+        #Other event properties?
+        
+        return res
+        
+    @staticmethod
+    def _config_as_selection_str(selection_string, data, pre_sel=None):
+        """Get mask for data base on the selection string"""
+        if pre_sel is None:
+            pre_sel = np.ones(len(data), dtype=np.bool_)
+
+        if selection_string != '':
+            if isinstance(selection_string, (list, tuple)):
+                selection_string = ' & '.join(f'({x})' for x in selection_string)
+
+            mask = numexpr.evaluate(selection_string, local_dict={
+                fn: data[fn]
+                for fn in data.dtype.names})
+            pre_sel &= mask
+        return pre_sel
+
+    def drift_time_R2_hist(self, data):
+        """Make area vs width 2D-hist"""
+        hist, _, _= np.histogram2d(
+            (data['s2_x'] ** 2 + data['s2_y'] ** 2),
+            data['drift_time'],
+            range=self.config['drift_time_vs_R2_bounds'],
+            bins=self.config['online_monitor_nbins'])
+        return hist.T
+
+
+
+
+
 class OnlineMonitor(strax.LoopPlugin):

```

Comment:
I think I'm going to delete this class. Why:
- It's slow to compute because it has this LoopPlugin structure, I think it's much easier to put the Online_event monitor and the online_peak monitor in separately.
- It makes the dependencies complicated, we already see this for the [online monitor as it is now
](https://straxen.readthedocs.io/en/latest/reference/datastructure.html#online-monitor), adding event level info will further amplify this. 
- It makes it easier to accept if either of the two is not available, that way you don't have to worry if we have some runs where events are not processed online. Also, you can process the online events later.